src/language/extension-types.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Extension types
+description: Learn how to write a static-only interface for an existing type.
+prevpage:
+  url: /language/extension-methods
+  title: Extension methods
+nextpage:
+  url: /language/callable-objects
+  title: Callable objects
+---
+
+An extension type is a compile time abstraction that "wraps"
+an existing type with a different, static-only interface.
+They allow you to enforce discipline on the set of operations available
+to objects of the underlying type, called the *representation type*.
+You can choose to use some members, omit others,
+replace others, and add new functionality, 
+all while getting compile time errors before runtime errors.
+
+The following example wraps the `int` type to create an extension type
+that only allows operations that make sense for ID numbers:
+
+```dart
+extension type IdNumber(int i) {
+  // Wraps the 'int' type's '<' operator:
+  operator <(IdNumber other) => i < other.i;
+  // Doesn't declare the '+' operator, for example,
+  // because addition does not make sense for ID numbers.
+}
+
+void main() {
+  // Without the discipline of an extension type,
+  // 'int' exposes ID numbers to unsafe operations:
+  int myUnsafeId = 42424242;
+  myUnsafeId = myUnsafeId + 10; // OK, but shouldn't be allowed for IDs.
+
+  var safeId = IdNumber(42424242);
+  safeId + 10; // Compile-time error, no '+' operator.
+  myUnsafeId = safeId; // Compile-time error, wrong type.
+  myUnsafeId = safeId as int; // OK, runtime cast to representation type.
+}
+```
+
+{{site.alert.note}}
+Extension types serve the same purpose as **wrapper classes**,
+but don't require the creation of expensive runtime objects.
+Because extension types are static-only and compiled away at runtime,
+they are essentially zero cost.
+
+[**Extension methods**]() are a static abstraction similar to extension types.
+However, an extension method adds functionality to *all* instances
+of its underlying type. Extension types are the opposite; 
+an extension type's interface only applies to objects
+declared with the extension type,
+and is distinct from the interface of its underlying type by default. 
+Read the [Transparency](#transparency) section for more details.
+{{site.alert.end}}
+
+## Syntax
+
+### Declaration
+
+Define a new extension type with the `extension type` declaration and a name,
+followed by the *representation type declaration* in parenthesis:
+
+```dart
+extension type E(int i) {
+  // Define set of operations
+}
+```
+
+The representation type declaration specifies that 
+the underlying type of extension type `E` is `int`,
+and the reference to the *representation object* is named `i`.
+
+The representation object gives access to the underlying type,
+the same way declaring a final instance variable of the underlying type
```

Comment:
I changed it to

> The representation object gives the extension type access to an object
typed as the representation type, the same way declaring the type of
a final instance variable as the underlying type would in a wrapper class.

But I'm starting to think removing the second clause might be the best solution because it's a confusing sentence no matter what. I initially included it to clarify the purpose for people who are familiar with wrapper classes, but maybe that's not valuable enough to warrant any possible confusion>